www/content/tutorial.md

```diff
@@ -1577,373 +1603,324 @@ Including the hash solves a number of problems:
+- Both the `!` suffix in the name, and the `=>` in the type, tell us this is an effectful function. Its two type parameters are just like the ones we saw in `Result` earlier: the first type tells us what this function will produce if it succeeds, and the other one tells us what it will produce if it fails.
+- `{}` tells us that this function always produces an empty record when it succeeds. (That is, it doesn't produce anything useful. Empty records don't have any information in them!) This is because the last expression in `main!` comes from `Stdout.line!`, which doesn't produce anything. (In contrast, the `Stdin` function's first type parameter is a `Str`, because it produces a `Str` if it succeeds.)
+- `[EndOfFile, StdinErr Stdin.IOErr, StdoutErr Stdout.IOErr]` tells us the different ways this function can fail. The `StdoutErr` and `StdinErr` tags are there because we used `Stdout.line!` and `Stdin.line!`.
 
-To understand what the `Exit I32 Str` error means, let's try temporarily commenting out our current `main` and replacing
+To understand the error a little more, let's try temporarily commenting out our current `main!` and replacing
 it with this one:
 
 ```roc
-main : Task {} [Exit I32 Str]
-main = Task.err (Exit 42 "An error happened!")
+main! : _ => Result {} [Exit I32 Str]
+main! = \_args -> Err (Exit 42 "An error happened!")
 ```
 
 Now if we run the application, it will print the line "An error happened!" to stderr and exit with a status code of 42. (You can check the status code of the most recent terminal command that finished in Windows by running `echo %ERRORLEVEL%` (or `$LASTEXITCODE` in PowerShell), or by running `echo $?` in macOS or Linux.)
 
 Now let's try running it with this version of `main`:
 
 ```roc
-main : Task {} [Exit I32 Str]
-main = Task.ok {}
+main! : _ => Result {} [Exit I32 Str]
+main! = \_args -> Ok {}
 ```
 
 This program won't print anything at all, but it will exit with a status code of `0`, indicating success.
 
 In summary:
 
-- If the `main` task ends in a `Task.ok {}`, then it means the final task succeeded and the program will exit with status code 0.
-- If the `main` task ends in a `Task.err (Exit 42 "…")`, then it means the final task failed, and the only information we got about the failure was that the program should exit with code 42 instead of 0, and that it should print a particular string to stderr to inform the user about what happened.
+- If the `main!` function ends in a `Ok {}`, then it means the final expression succeeded and the program will exit with status code 0.
+- If the `main!` function ends in a `Err (Exit 42 "…")`, then it means it failed, and the only information we got about the failure was that the program should exit with code 42 instead of 0, and that it should print a particular string to stderr to inform the user about what happened.
+
+### [Handling failure](#handling-failures) {#handling-failures}
```

Comment:
I suggest to either add the final `s` to the title or remove it from the identifier.

Reply:
Done in 53f5fc3e79